### PR TITLE
Cast tag option

### PR DESCRIPTION
**Pull request by rvolosatovs**
The `structs` package does not provide all the necessary functionality - I used `reflect` directly

### CODE REVIEW

File: encoding/encode.go [hunk 1]
```diff
@@ -209,63 +219,112 @@ func ToStringStringMap(tagName string, input interface{}, properties ...string)
...
+				var v uint16
+				t = reflect.TypeOf(v)
+			case "uint8":
+				var v uint8
+				t = reflect.TypeOf(v)
+			case "string":
+				var v string
+				t = reflect.TypeOf(v)
+			default:
+				panic(fmt.Errorf("Wrong cast type specified: %d", opts.Value("cast")))
```
Comment (htdvisser) [inline encoding/encode.go]: Looks like a string to me, not `%d`

File: encoding/encoding_test.go [hunk 2]
```diff
@@ -728,5 +728,20 @@ func TestNonUniqueKeys(t *testing.T) {
...
+
+func TestCastType(t *testing.T) {
+	a := s.New(t)
 
+	strct := struct {
+		Int64 uint64 `test:"i64,cast=int64"`
+	}{
+		42,
+	}
+
```
Comment (htdvisser) [inline encoding/encoding_test.go]: - Please test for the actual value, the result could currently still be a nice `int64` of `0`
- Please test for error cases. What if you try to convert an int to a string? What if you try to convert other types?

**Comment by rvolosatovs** [comment]
I don't think `cast` name fits this option anymore.
`int64` can not be casted to `string`, yet this option will convert it to `string` form.

This should also be included in decoding, I think.
That would eliminate the issue with Influx, where `uint64` are stored as strings, for example.

I removed `structs`, because the package does not expose the `reflect.Value` of the value. It is not possible to call `Convert` on the value returned - it is required to reflect on the value returned by the `structs` package, which means using reflection twice, which is inefficient and unnecessary. Using `reflect` directly gives us much more flexibility.

**Comment by rvolosatovs** [comment]
@htdvisser

**Comment by htdvisser** [comment]
As the `structs` package does not contain the functionality needed, the `encoding` package will first be refactored to use `reflect` from the Go standard library: https://github.com/TheThingsNetwork/go-utils/issues/31

**Pull request closed without merging**